### PR TITLE
MW-129: Introduced JaVers to all domain entities.

### PR DESCRIPTION
**Pull request by lukaslew**

### CODE REVIEW

**Comment by pgesek** [comment]
Well first of all I don't see how this relates to the ticket, but I suppose it makes sense to keep an audit trail for all of ref data entities.

File: src/integration-test/java/org/openlmis/referencedata/web/GeographicZoneControllerIntegrationTest.java [hunk 1]
```diff
@@ -465,6 +467,64 @@ public void findByLocationShouldReturnForbiddenForUnauthorizedToken() {
...
+    restAssured
+        .given()
+        .header(HttpHeaders.AUTHORIZATION, getTokenHeader())
+        .pathParam("id", UUID.randomUUID())
+        .when()
+        .get(AUDIT_URL)
+        .then()
+        .statusCode(200);
+
+    assertThat(RAML_ASSERT_MESSAGE, restAssured.getLastReport(), RamlMatchers.hasNoViolations());
```
Comment (pgesek) [inline src/integration-test/java/org/openlmis/referencedata/web/GeographicZoneControllerIntegrationTest.java]: can we put these tests methods in the base class? From what I can see the only differences across entities are the base of the url, the right name and the 1 line of mocking done beforehand.

File: src/main/java/org/openlmis/referencedata/web/FacilityController.java [hunk 2]
```diff
@@ -164,7 +169,13 @@ public String getFacilitiesAuditLog(
       throw new NotFoundException(FacilityMessageKeys.ERROR_NOT_FOUND);
     }
 
-    return getAuditLog(Facility.class, id, author, changedPropertyName, page, returnJson);
+    String auditLogs = getAuditLog(Facility.class, id, author, changedPropertyName, page,
```
Comment (pgesek) [inline src/main/java/org/openlmis/referencedata/web/FacilityController.java]: these 4 lines can be get bundled into one method reused across controllers

File: src/integration-test/java/org/openlmis/referencedata/web/GeographicZoneControllerIntegrationTest.java (hunk 1 above)
Comment (lukaslew) [inline src/integration-test/java/org/openlmis/referencedata/web/GeographicZoneControllerIntegrationTest.java]: Done

File: src/main/java/org/openlmis/referencedata/web/FacilityController.java (hunk 2 above)
Comment (lukaslew) [inline src/main/java/org/openlmis/referencedata/web/FacilityController.java]: Done

File: src/integration-test/java/org/openlmis/referencedata/web/AuditLogWebIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+        messageKey,
+        isOneOf(MESSAGEKEY_ERROR_UNAUTHORIZED, MESSAGEKEY_ERROR_UNAUTHORIZED_GENERIC)
+    );
+    assertThat(RAML_ASSERT_MESSAGE, restAssured.getLastReport(), RamlMatchers.hasNoViolations());
+  }
+
+  @Test
+  public void shouldGetAuditLog() {
+    mockHasAuditRight();
+    given(getRepository().findOne(any(UUID.class))).willReturn(getInstance());
```
Comment (pgesek) [inline src/integration-test/java/org/openlmis/referencedata/web/AuditLogWebIntegrationTest.java]: I was thinking more that each test can do this mocking line on its own, then call a method called testAuditLog() or something. AUDIT_LOG is also always the same, testAuditLog() could take the base url as param - currently there is I think too much boilerplate to implement in every test.

Comment (lukaslew) [inline src/integration-test/java/org/openlmis/referencedata/web/AuditLogWebIntegrationTest.java]: Done(?)

**Comment by joshzamor** [comment]
@lukaslew:  it looks like this one needs to be updated before it can be merged, and that @pgesek has changes requested that are still outstanding.  Could you update and keep this moving forward?

**Comment by lukaslew** [comment]
@joshzamor I resolved conflicts.
@pgesek I made requested changes. Are we okay now?

**Comment by pgesek** [comment]
Yeah, here goes

**Comment by joshzamor** [comment]
Merging this in appeared to break demo-data:  http://build.openlmis.org/job/OpenLMIS-requisition-contract-test/1801/

I'm going to revert this, re-open the branch and ask that we try again.

**Comment by lukaslew** [comment]
@pgesek @joshzamor I created new PR: https://github.com/OpenLMIS/openlmis-referencedata/pull/25 This time I checked with contract-tests for the requisition service and they passed on my machine.

**Pull request merged**